File: esoteric_entropy/platform.py
"""Platform detection — discover available entropy sources."""

from __future__ import annotations

import platform as _platform

from esoteric_entropy.sources import ALL_SOURCES
from esoteric_entropy.sources.base import EntropySource


def detect_available_sources() -> list[EntropySource]:
    """Instantiate and return all sources available on this machine."""
    available: list[EntropySource] = []
    for cls in ALL_SOURCES:
        try:
            src = cls()
            if src.is_available():
                available.append(src)
        except Exception:
            continue
    return available


def platform_info() -> dict:
    """Return basic platform metadata."""
    return {
        "system": _platform.system(),
        "machine": _platform.machine(),
        "platform": _platform.platform(),
        "python": _platform.python_version(),
    }

File: crates/esoteric-core/src/platform.rs
//! Platform detection and source discovery.

use crate::source::EntropySource;
use crate::sources::all_sources;

/// Discover all entropy sources available on this machine.
pub fn detect_available_sources() -> Vec<Box<dyn EntropySource>> {
    all_sources()
        .into_iter()
        .filter(|s| s.is_available())
        .collect()
}

/// Platform information.
pub fn platform_info() -> PlatformInfo {
    PlatformInfo {
        system: std::env::consts::OS.to_string(),
        machine: std::env::consts::ARCH.to_string(),
        family: std::env::consts::FAMILY.to_string(),
    }
}

#[derive(Debug, Clone)]
pub struct PlatformInfo {
    pub system: String,
    pub machine: String,
    pub family: String,
}
